src/main/scala/uclid/lang/UclidLanguage.scala

```diff
@@ -107,6 +107,21 @@ object ULContext {
     }
     recordTypes.exists(r => r.fields.map(_._1).contains(id))
   }
+  // Leiqi: add checkField
+  def checkuninterpretedField(id : Identifier) : Boolean = {
+    val recordTypes = origTypeMap.fwdMap.map(_._2) flatMap {
+      case r : RecordType => Some(r)
+      case _ => None
+    }
+    val identifierName = id.toString()
+    val index = identifierName.indexOf("!")
+    val FieldName = identifierName.substring(0,index)
+    val tempIdentifier = Identifier(FieldName)
+    //we get utype out of utype!val!0 and make a identifier with name "utype"
+    //then we search the existence of Identifier "utype"
+    recordTypes.exists(r => r.fields.map(_._1).contains(tempIdentifier))
```

Comment:
And corresponding change here:
```suggestion
    uninterpretedTypes.exists(r => r.name.equals(tempIdentifier))
```